File: leetcode/C++/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int idx=0,n=1,mIdx=0,mn=0;
        sort(nums.begin(), nums.end());
        for(int i=1; i<nums.size(); ++i){
            if(nums[i]==nums[i-1]) {
                if(mn<++n) {mIdx=idx; mn=n;}
            }
            else {idx=i; n=1;}
        }
        return nums[mIdx];
    }
};

File: leetcode/Python/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        mn, n, mIdx, idx = 0, 1, 0, 1
        nums.sort()
        for i in range(1,len(nums)):
            if nums[i]==nums[i-1]:
                n+=1
                if n > mn:
                    mn, mIdx = n, idx
            else:
                n, idx = 1, i
        return nums[mIdx]